support/sier-codec/src/schema.rs

```diff
@@ -75,16 +75,57 @@ impl FieldDef {
             Type::U8 => complete::le_u8(bytes).map(|(b, n)| (b, Value::U8(n))),
             Type::U32 => complete::le_u32(bytes).map(|(b, n)| (b, Value::U32(n))),
             Type::U64 => complete::le_u64(bytes).map(|(b, n)| (b, Value::U64(n))),
+            Type::String => {
+                let (bytes, str_bytes) = length_prefixed(bytes).map_err(Error::ValueParsing)?;
+                let s = std::str::from_utf8(str_bytes).map_err(Error::InvalidUtf8)?;
+
+                Ok((bytes, Value::String(String::from(s))))
+            }
         }
         .map_err(Error::ValueParsing)
     }
 }
 
+use nom::{
+    bytes::complete::{take, take_while},
+    combinator::recognize,
+    sequence::pair,
+    IResult,
+};
+
+fn length_prefixed(b: &[u8]) -> IResult<&[u8], &[u8]> {
+    let (b, len) = var_int(b)?;
+    take(len)(b)
+}
+
+fn var_int(b: &[u8]) -> IResult<&[u8], usize> {
+    let (new_b, int_bytes) = recognize(pair(take_while(|b| b & 128 > 0), take(1usize)))(b)?;
```

Comment:
Just to make sure I get it right: our strings are length prefixed, var int at the header (128 bits). We're shifting that byte by byte until we extract the numeric length value and the remainder which is the string. We then extract that length from the buffer.

Is this used anywhere else than strings?

Reply:
The "varint" representation will be used in a lot of places. It's used for any length for the codec (like length of strings or length of lists). It can also be the representation of numbers that are disproportionately distributed near 0. Varints are the default for numbers in protobuf. SCALE also has a similar approach to reducing the byte-size of small-ish numbers.